internal/weldr/api.go

```diff
@@ -1883,6 +1915,31 @@ func (api *API) composeHandler(writer http.ResponseWriter, request *http.Request
 		return
 	}
 
+	// Fetch parent ostree commit from ref + url if commit is not
+	// provided. The parameter name "parent" is perhaps slightly misleading
+	// as it represent whatever commit sha the image type requires, not
+	// strictly speaking just the parent commit.
+	if cr.OSTree.Ref != "" && cr.OSTree.URL != "" {
+		if cr.OSTree.Parent != "" {
+			errors := responseError{
+				ID:  "OSTreeOptionsError",
+				Msg: "Supply at most one of Parent and URL",
+			}
+			statusResponseError(writer, http.StatusInternalServerError, errors)
```

Comment:
This should imho be `http.BadRequestError`

Reply:
`http.StatusBadRequest`, but yes, agreed. And fixed.